eng/Directory.Build.Common.props

```diff
@@ -125,8 +125,8 @@
     <IsPackable>false</IsPackable>
     <!-- List newest targets first so that recordings are made with latest, running tests from editor runs latest, etc. -->
     <RequiredTargetFrameworks>net7.0;net6.0</RequiredTargetFrameworks>
-    <!-- Also test net461 on Windows. -->
-    <RequiredTargetFrameworks Condition="'$(OS)' == 'Windows_NT'">$(RequiredTargetFrameworks);net461</RequiredTargetFrameworks>
+    <!-- Also test net462 on Windows. net461 is out of support as of 4/2022 and is not supported in version 4.4.x and beyond of NUnit3TestAdapter. -->
```

Comment:
Are there other places that might need to change as well?

For example:
- https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/perf/EventSourceBenchmark.cs#L17
- https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/tests/HttpClientTransportFunctionalTest.cs#L31
- https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Azure.Core.csproj#L36

Reply:
We don't want to change any product code, but there may be some places in tests that need updating if they previously referenced net461 specifically.

Reply:
Yeah, I don't have a good understanding here at all, I just noticed there's a lot of hits for `net461` in our repo:
https://grep.app/search?q=net461&filter[repo][0]=Azure/azure-sdk-for-net&filter[path][0]=sdk/core/

Reply:
Yep, I will go ahead and make all the test updates once we get feedback on whether the approach here makes sense.

Reply:
Sounds good - thanks for driving this!

Reply:
Looks like I will need to update these as well - https://grep.app/search?q=TestTargetFramework&filter[repo][0]=Azure/azure-sdk-for-net

Reply:
> I think the approach makes sense. We should also remove the `netcoreapp3.1` target from those pipeline configurations, as it is also EOL and unsupported.

I'd rather leave that on for now at least so that we are not reducing the number of test runs. Also, this could have an impact on the split pipelines for Core. I will file this as a separate issue.

Reply:
> Do we have a sense of where we have the most customers today? I understand the EOL being one data point to consider for our test matrix however if a large majority of our customers are on net461 we should continue to test on that version.
> 
> Also does this change affect mgmt SDKs too? Do we know what the customer distribution is there? We should at least be consistent in our support matrix across data plane and mgmt plane.

This would apply to all T2 libraries. net461 is EOL so it won't even get security updates. I don't think we can realistically avoid upgrading the NUnit3TestAdapter, because as @annelo-msft found, we need the later versions to support newer .NET versions.

Reply:
>  if a large majority of our customers are on net461 we should continue to test on that version.

I respectfully disagree.  If the framework is out of support, there is no support for issues specific to that framework - we wouldn't release a patch for it, we would advise moving to a supported framework version. 

It's also not advised that we run unsupported frameworks in our pipelines.  They no longer receive security fixes and could open us to a vulnerability.

Reply:
I did a grep on the SDK code and see a few hits for special casing net461? Should we look at those to see if they are applicable anymore? Given it affects all T2 libraries make sure we get an ack from Arthur Ma as well. I think we should be consistent in our support matrix for data plane as well as mgmt plane. 

> It's also not advised that we run unsupported frameworks in our pipelines. They no longer receive security fixes and could open us to a vulnerability.

I have no concerns with the switch per se but this makes me ask what is our SDK support policy and why?

Reply:
[Azure SDK Support Policy](https://azure.github.io/azure-sdk/policies_support.html)

At a high-level:  we do not support products that are out of support.   This includes frameworks that our code may run and other dependencies.

Reply:
/cc: @KrzysztofCwalina, @tg-msft, @scottaddie as FYI